force-app/main/default/lwc/refundPayment/refundPayment.js

```diff
@@ -0,0 +1,132 @@
+import { LightningElement, api, wire } from 'lwc';
+import { CloseActionScreenEvent } from 'lightning/actions';
+import { getRecord, getFieldValue } from "lightning/uiRecordApi";
+import { constructErrorMessage} from "c/utilCommon";
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import refundPaymentTitle from "@salesforce/label/c.pmtRefundPaymentTitle";
+import cancelButtonLabel from "@salesforce/label/c.stgBtnCancel";
+import commonRefreshPage from "@salesforce/label/c.commonRefreshPage";
+import noRefundPermissionMessage from "@salesforce/label/c.pmtNoRefundPermissionMessage";
+import refundPaymentErrorMessage from "@salesforce/label/c.pmtRefundPaymentErrorMessage";
+import refundPaymentMessage from "@salesforce/label/c.pmtRefundPaymentMessage";
+import refundProcessing from "@salesforce/label/c.pmtRefundProcessing";
+
+import processRefund from "@salesforce/apex/PMT_RefundController.processRefund";
+import getPermissionData from "@salesforce/apex/PMT_RefundController.getPermissionData";
+
+import PAYMENT_AMOUNT_FIELD from '@salesforce/schema/npe01__oppPayment__c.npe01__Payment_Amount__c';
+import PAYMENT_DATE_FIELD from '@salesforce/schema/npe01__oppPayment__c.npe01__Payment_Date__c';
+
+export default class Lwc_QuickAction extends LightningElement {
+    @api recordId;
+    hasError = false;
+    errorMessage;
+    labels = Object.freeze({
+        refundPaymentTitle,
+        cancelButtonLabel,
+        commonRefreshPage,
+        noRefundPermissionMessage,
+        refundPaymentMessage,
+        refundProcessing,
+        refundPaymentErrorMessage
+    });
+    refundView;
+    paymentAmount;
+    _paymentDate;
+
+    @wire(getRecord, {
+        recordId: '$recordId',
+        fields: [PAYMENT_AMOUNT_FIELD, PAYMENT_DATE_FIELD]})
```

Comment:
Do we think we can enrich the `refundView` with this data? That way it is truly the source of truth for all things `view` for refunds and we're retrieving data in two different ways for refunds.

Reply:
That was my original plan. However, there is an issue with Quick Action in LWC where the `recordId` is not accessible in `ConnectCallback` function. Currently, the only other solution I found is to move the apex class to `RenderCallback` and have a boolean control to make the server call, which I think it's kind of like an anti-pattern. I will continue the research to see if there is any other viable way.